Segmentation fault when calling Go func

The below code causes a panic due to a null pointer dereference in goContext() (ctx returned by getContext() seems nil).

```go
package main

import (
	"fmt"

	"rogchap.com/v8go"
)


func main() {
	iso, _ := v8go.NewIsolate()
	global, _ := v8go.NewObjectTemplate(iso)

	printfn, _ := v8go.NewFunctionTemplate(iso, func(info *v8go.FunctionCallbackInfo) *v8go.Value {
		fmt.Printf("%+v\n", info.Args())
		return nil
	})

	global.Set("print", printfn, v8go.ReadOnly)


	ctx, _ := v8go.NewContext(iso, global)
	val, _ := ctx.RunScript(`() => { print("foo"); }`, "value.js")
	fn, _ := val.AsFunction()
	val, _ = fn.Call()

	fmt.Printf("result: %+v\n", val)
}
```

```
$ go run test.go 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x850ed3]

goroutine 1 [running]:
rogchap.com/v8go.goContext(...)
	/home/user/go/pkg/mod/rogchap.com/v8go@v0.5.2-0.20210410040105-4c18ca58a475/context.go:164
rogchap.com/v8go._Cfunc_FunctionCall(0x2c2f3e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	_cgo_gotypes.go:191 +0x4d
rogchap.com/v8go.(*Function).Call.func1(0xc000067f40, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc000067e78, 0xc000067ea0, 0x850d15)
	/home/user/go/pkg/mod/rogchap.com/v8go@v0.5.2-0.20210410040105-4c18ca58a475/function.go:28 +0x93
rogchap.com/v8go.(*Function).Call(0xc000067f40, 0x0, 0x0, 0x0, 0x8, 0xc000012320, 0x0)
	/home/user/go/pkg/mod/rogchap.com/v8go@v0.5.2-0.20210410040105-4c18ca58a475/function.go:28 +0x11e
main.main()
	/home/user/dev/foo/test.go:25 +0x1c5
exit status 2
```

--- a/function.go
+++ b/function.go
@@ -25,6 +25,8 @@ func (fn *Function) Call(args ...Valuer) (*Value, error) {
 		}
 		argptr = (*C.ValuePtr)(unsafe.Pointer(&cArgs[0]))
 	}
+	fn.ctx.register()
 	rtn := C.FunctionCall(fn.ptr, C.int(len(args)), argptr)
+	fn.ctx.deregister()
 	return getValue(fn.ctx, rtn), getError(rtn)
 }

--- a/function_test.go
+++ b/function_test.go
@@ -33,6 +33,37 @@ func TestFunctionCall(t *testing.T) {
 	}
 }
 
+func TestFunctionCallToGoFunc(t *testing.T) {
+	t.Parallel()
+
+	iso, _ := v8go.NewIsolate()
+	global, _ := v8go.NewObjectTemplate(iso)
+
+	called := false
+	printfn, _ := v8go.NewFunctionTemplate(iso, func(info *v8go.FunctionCallbackInfo) *v8go.Value {
+		called = true
+		return nil
+	})
+
+	global.Set("print", printfn, v8go.ReadOnly)
+
+	ctx, err := v8go.NewContext(iso, global)
+	failIf(t, err)
+	val, err := ctx.RunScript(`(a, b) => { print("foo"); }`, "")
+	failIf(t, err)
+	fn, err := val.AsFunction()
+	failIf(t, err)
+	resultValue, err := fn.Call()
+	failIf(t, err)
+
+	if !called {
+		t.Errorf("expected my function to be called, wasn't")
+	}
+	if !resultValue.IsUndefined() {
+		t.Errorf("expected undefined, got: %v", resultValue.DetailString())
+	}
+}
+
 func TestFunctionCallError(t *testing.T) {
 	t.Parallel()
 

